### PR TITLE
Fix env data tests

### PR DESCRIPTION
**Pull request by belfortep**
Closes https://github.com/lambdaclass/evm_mlir/issues/386
## Summary
This PR fix an issue with the parsing of the transaction data of EF tests
## Changes
- Use of the function to parse from the hex string to hex value
- Updated the list of ignored group of tests
- Small fix to the `call` syscall, using `checked_sub` so it doesn't panic

### CODE REVIEW

File: tests/ef_tests.rs [hunk 1]
```diff
@@ -96,6 +91,10 @@ fn get_ignored_groups() -> HashSet<String> {
         "stExample".into(),
         "vmArithmeticTest".into(),
         "stQuadraticComplexityTest".into(),
+        "stSelfBalance".into(),
+        "stEIP3855-push0".into(),
+        "stWalletTest".into(),
+        "vmLogTest".into(),
```
[the remark this answers is not in the document]
Comment (belfortep) [reply tests/ef_tests.rs:97]: Yes, these tests are broken now

[the remark this answers is not in the document]
Comment (juanimedone) [reply tests/ef_tests.rs:97]: So they were just randomly succeeding before, right?

[the remark this answers is not in the document]
Comment (belfortep) [reply tests/ef_tests.rs:97]: Yes, when they access their data they were getting unexpected things that could or not work

**Review by juanimedone — APPROVED** [review]
LGTM!

File: tests/ef_tests.rs [hunk 2]
```diff
@@ -152,7 +151,7 @@ fn run_test(path: &Path, contents: String) -> datatest_stable::Result<()> {
             env.tx.caller = sender;
             env.tx.gas_limit = unit.transaction.gas_limit[test.indexes.gas].as_u64();
             env.tx.value = unit.transaction.value[test.indexes.value];
-            env.tx.data = unit.transaction.data[test.indexes.data].clone();
+            env.tx.data = convert_to_hex(unit.transaction.data[test.indexes.data].clone());
```
Comment (MegaRedHand) [inline tests/ef_tests.rs:154]: Can we rename this function? `convert_to_hex` makes me think it turns bytes to their hex representation when it does the opposite.

File: tests/ef_tests.rs [hunk 3]
```diff
@@ -194,15 +193,30 @@ fn run_test(path: &Path, contents: String) -> datatest_stable::Result<()> {
...
+                            return Err("Wrong output".into());
+                        }
+                    }
+                }
+                (
+                    Some(_),
+                    ExecutionResult::Halt {
+                        reason: _,
+                        gas_used: _,
+                    },
```
Comment (MegaRedHand) [inline tests/ef_tests.rs:211]: You can use `..` instead of listing all fields explicitly

File: tests/ef_tests.rs [hunk 4]
```diff
@@ -194,15 +193,30 @@ fn run_test(path: &Path, contents: String) -> datatest_stable::Result<()> {
...
+                        unit.out.as_ref().zip(res.result.output())
+                    {
+                        if expected_output != output {
+                            return Err("Wrong output".into());
+                        }
+                    }
+                }
+                (
+                    Some(_),
+                    ExecutionResult::Halt {
```
Comment (MegaRedHand) [inline tests/ef_tests.rs:208]: Are we sure this doesn't include reverts?

File: tests/ef_tests.rs [hunk 5]
```diff
@@ -194,15 +193,30 @@ fn run_test(path: &Path, contents: String) -> datatest_stable::Result<()> {
...
-            if test.expect_exception.is_some() {
-                assert!(!res.result.is_success());
-                // NOTE: the expect_exception string is an error description, we don't check the expected error
-                continue;
+            match (&test.expect_exception, &res.result) {
+                (None, _) => {
+                    if let Some((expected_output, output)) =
+                        unit.out.as_ref().zip(res.result.output())
+                    {
+                        if expected_output != output {
```
Comment (MegaRedHand) [inline tests/ef_tests.rs:201]: Isn't this ignoring output mismatches?
I think we can simplify this to:

```suggestion
                    if unit.out.as_ref() != res.result.output() {
```

File: tests/ef_tests.rs (hunk 2 above)
Comment (belfortep) [reply tests/ef_tests.rs]: Oh that's true, it takes the hex and converts to decimal. Changed the name and added a little documentation

File: tests/ef_tests.rs (hunk 4 above)
Comment (belfortep) [reply tests/ef_tests.rs]: You are right, it should have the `Revert` enum option also

File: tests/ef_tests.rs [hunk 6]
```diff
@@ -34,12 +34,11 @@ fn get_ignored_groups() -> HashSet<String> {
         "stEIP5656-MCOPY".into(),
         "stEIP3651-warmcoinbase".into(),
         "stArgsZeroOneBalance".into(),
-        "stTimeConsuming".into(),
+        //"stTimeConsuming".into(), // this works, but it's REALLY is time consuming
```
Comment (juanimedone) [inline tests/ef_tests.rs:37]:
```suggestion
        //"stTimeConsuming".into(), // this works, but it's REALLY time consuming
```

File: tests/ef_tests.rs [hunk 7]
```diff
@@ -105,8 +105,11 @@ fn get_ignored_suites() -> HashSet<String> {
     ])
 }
 
-fn convert_to_hex(account_info_code: Bytes) -> Bytes {
-    let hex_string = std::str::from_utf8(&account_info_code[2..]).unwrap(); // we don't need the 0x
+/// Receives a Bytes object with the hex representation
+/// And returns a Bytes object with the decimal representation
+/// Taking the hex numbers by pairs
+fn bytes_in_hex_to_decimal(bytes_in_hex: Bytes) -> Bytes {
+    let hex_string = std::str::from_utf8(&bytes_in_hex[2..]).unwrap(); // we don't need the 0x
```
Comment (MegaRedHand) [inline tests/ef_tests.rs:112]: Let's check this is `0x` and fail if not.

File: tests/ef_tests.rs [hunk 8]
```diff
@@ -105,8 +105,11 @@ fn get_ignored_suites() -> HashSet<String> {
     ])
 }
 
-fn convert_to_hex(account_info_code: Bytes) -> Bytes {
-    let hex_string = std::str::from_utf8(&account_info_code[2..]).unwrap(); // we don't need the 0x
+/// Receives a Bytes object with the hex representation
+/// And returns a Bytes object with the decimal representation
+/// Taking the hex numbers by pairs
+fn bytes_in_hex_to_decimal(bytes_in_hex: Bytes) -> Bytes {
```
Comment (MegaRedHand) [inline tests/ef_tests.rs:111]: How about `decode_hex` instead?

File: tests/ef_tests.rs (hunk 7 above)
Comment (belfortep) [reply tests/ef_tests.rs:112]: Should we change to return an Option or just panics?

**Review by MegaRedHand — APPROVED** [review]
LGTM

**Pull request merged**